juego.c

```diff
@@ -0,0 +1,324 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <string.h>
+#include <windows.h>
+
+int main()
+{
+    char tablero[10][10], tableroVisible[10][10], comando, celda, reiniciar;
+    int jugadorX, jugadorY, tesorosRestantes, numTrampas, numTesoros, numPistas;
+    int i, j, x, y, distancia, nuevaX, nuevaY, juegoTerminado, distTemp;
+    HANDLE hConsole;
+
+    reiniciar = 's';
+    hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
+
+    while (reiniciar == 's' || reiniciar == 'S')
+    {
+        srand(time(NULL));
+        for (i = 0; i < 10; i++)
+        {
+            for (j = 0; j < 10; j++)
+            {
+                tablero[i][j] = ' ';
+                tableroVisible[i][j] = ' ';
+            }
+        }
+
+        system("cls");
+        SetConsoleTextAttribute(hConsole, 11);
+        printf("\n******************************************\n");
+        printf("*  BIENVENIDO AL JUEGO DE BUSQUEDA DEL   *\n");
+        printf("*               TESORO                   *\n");
+        printf("******************************************\n\n");
+        SetConsoleTextAttribute(hConsole, 7);
+
+        printf("Configuracion inicial:\n");
+        printf("Numero de trampas: ");
+        scanf("%d", &numTrampas);
+        printf("Numero de tesoros: ");
+        scanf("%d", &numTesoros);
+        printf("Numero de pistas: ");
+        scanf("%d", &numPistas);
+
+        jugadorX = rand() % 10;
+        jugadorY = rand() % 10;
+
+        tesorosRestantes = numTesoros;
+        for (i = 0; i < numTesoros; i++)
+        {
+            do
+            {
+                x = rand() % 10;
+                y = rand() % 10;
+            }
+            while ((x == jugadorX && y == jugadorY) || tablero[x][y] != ' ');
+            tablero[x][y] = 'T';
+        }

```

Comment:
Encuentro raro que hay dos ciclos aqui (Un `do` `while` dentro del `for`)

Que quieren lograr?